### ISSUE TITLE
 Wrong Decoding of Mapschema with number numeric string keys in C# client

### ISSUE THREAD

**Issue by roberto-proenca** [colyseus/schema#42]
when we create a mapschema with numeric indexes and try to decode it in the C# client the updated values are called with the wrong keys!

i created a repository to [showcase](https://github.com/roberto-proenca/bug-colyseus-schema) this bug.

in it the server starts a  mapschema with the values: 

```javascript
items = {"bbb":1, "aaa":1, "221":1, "021":1, "15": 1, "10":1}
```

it is initialised correctly in the unity client, but when we try to update a value in the client, another one is updated in the client:

for example, if we set 
```javascript
items["10"] = 2
```
in the client the value updated will be `items["bbb"]` with value `2`

i this problem occurs because the server serialiser depends on object iteration order in javascript, and objects in  javascript have a strange iteration order ([link1](https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/), [link2](https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/))

so in our case:
the client starts the object with correct order:

```
items = {"bbb": 1, "aaa": 1, "221": 1, "021": 1, "15": 1, "10": 1}
```

but in the server the indexes are reordered to: 
```
items = {"10": 11, "15": 1, "221": 1, "bbb": 1, "aaa": 1, "021": 1}
```

so when we update the `items["10"] = 2` the client receives something like:
update the object at index 0, that in the server is "10"(javascript order) but in the client is "bbb"(insertion order)

one possible solution in my opinion is tho keep the keys and values in a proper Map instead of an object, so the insertion order is guaranteed

**Comment by endel** [comment]
Thank you so much for the detailed report @roberto-proenca. I've managed to fix this on version ` @colyseus/schema@0.4.58`. The encoder is now considering insertion order instead of the JavaScript's default ordering.

I've added a test for this here: https://github.com/colyseus/colyseus-unity3d/commit/6963578026666f1f2ac78cc5e3cdeb0bf9790928

To update you can remove your `node_modules` + `package-lock.json` and install them again! Let me know if it works for you. Cheers!

**Comment by roberto-proenca** [comment]
Wow, thank you, i just tested it here and it seems to be working fine!

Next week i plan to investigate some other problems i ran into in my test project, i hope i can count with your help again :)

since it is behaving correctly now, i will close the issue!

thank you!
